### PR TITLE
Move private APIs to the private folder.

### PR DESCRIPTION
**Pull request by jverkoey**
This will allow the APIs to be reused across source files within the project.

### CODE REVIEW

File: src/MDMMotionAnimator.m [hunk 1]
```diff
@@ -16,24 +16,11 @@
...
-}
+#import "private/CABasicAnimation+MotionAnimator.h"
+#import "private/MDMUIKitValueCoercion.h"
+#import "private/MDMDragCoefficient.h"
 
-static CAMediaTimingFunction* timingFunctionWithControlPoints(CGFloat controlPoints[4]);
-static NSArray* coerceUIKitValuesToCoreAnimationValues(NSArray *values);
-static CABasicAnimation *animationFromTiming(MDMMotionTiming timing, CGFloat timeScaleFactor);
-static void makeAnimationAdditive(CABasicAnimation *animation);
+#import <UIKit/UIKit.h>
```
Comment (romoore) [inline src/MDMMotionAnimator.m]: Format/Nit: UIKit should be moved up to just after the related header.

File: src/private/CABasicAnimation+MotionAnimator.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+#import <QuartzCore/QuartzCore.h>
+
+// Returns a basic animation configured with the provided timing and scale factor.
+FOUNDATION_EXPORT
+CABasicAnimation *MDMAnimationFromTiming(MDMMotionTiming timing, CGFloat timeScaleFactor);
+
+// Attemps to configure the provided animation to be additive.
+// Not all animation value types are supported. If an animation's value type was not supported,
+// the animation will not be modified.
+FOUNDATION_EXPORT void MDMMakeAnimationAdditive(CABasicAnimation *animation);
```
Comment (romoore) [inline src/private/CABasicAnimation+MotionAnimator.h:30]: As a follow-up, can this return a BOOL to determine success?

File: src/private/MDMUIKitValueCoercion.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+// Coerces the following UIKit values to Core Animation values:
+//
+// - UIBezierPath -> CGPath
+// - UIColor -> CGColor
+FOUNDATION_EXPORT NSArray* MDMCoerceUIKitValuesToCoreAnimationValues(NSArray *values);
```
Comment (romoore) [inline src/private/MDMUIKitValueCoercion.h]: Maybe document that the array must contain values of all the same type, mixed arrays will fail.

**Review by romoore — APPROVED** [review]
Generally looks good. Please address comments before merging.

File: src/MDMMotionAnimator.m (hunk 1 above)
Comment (jverkoey) [reply src/MDMMotionAnimator.m]: Done.

File: src/private/MDMUIKitValueCoercion.h (hunk 3 above)
Comment (jverkoey) [reply src/private/MDMUIKitValueCoercion.h]: Done.

File: src/private/CABasicAnimation+MotionAnimator.h (hunk 2 above)
Comment (jverkoey) [reply src/private/CABasicAnimation+MotionAnimator.h:30]: Follow-up filed here: https://github.com/material-motion/motion-animator-objc/issues/21

**Pull request merged**